### PR TITLE
server: return a better error when container requests pod pid ns, but…

### PR DESCRIPTION
**Pull request by haircommander**
… infra not created

Signed-off-by: Peter Hunt <pehunt@redhat.com>

#### What type of PR is this?

/kind bug
#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

File: server/container_create_linux.go [hunk 1]
```diff
@@ -425,7 +425,10 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 	} else if securityContext.NamespaceOptions.Pid == types.NamespaceModePOD {
 		pidNsPath := sb.PidNsPath()
 		if pidNsPath == "" {
-			return nil, errors.New("PID namespace requested, but sandbox infra container invalid")
+			if sb.NamespaceOptions().Pid != types.NamespaceModePOD {
+				return nil, errors.New("Pod level PID namespace requested for container, but pod sandbox was not similarly configured, and does not have an infra container.")
```
Comment (TomSweeneyRedHat) [inline server/container_create_linux.go:429]:
```suggestion
				return nil, errors.New("Pod level PID namespace requested for the container, but pod sandbox was not similarly configured and does not have an infra container.")
```

**Comment by TomSweeneyRedHat** [comment]
Couple nit suggestions, otherwise LGTM once tests are hip

File: server/container_create_linux.go [hunk 2]
```diff
@@ -425,7 +425,10 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
...
+			}
+			return nil, errors.New("PID namespace requested, but sandbox infra container unexpectedly invalid.")
```
Comment (saschagrunert) [inline server/container_create_linux.go:431]:
```suggestion
			return nil, errors.New("PID namespace requested, but sandbox infra container unexpectedly invalid")
```

File: server/container_create_linux.go (hunk 1 above)
Comment (saschagrunert) [reply server/container_create_linux.go:429]: Please also remove the dot from the error message at the end.

**Review by mrunalp — APPROVED** [review]
/lgtm

**Pull request merged**